### DISCUSSION TITLE
Surrogate Modelling

### DISCUSSION THREAD

**Discussion by GForb** [pmsims-package/pmsims#10] [Ideas]
This is a thread to discuss issues relating to the implementation of the surrogate modelling in this project. 

## Aim
The aim of the surrogate modelling is to estimate the curve of predictive performance vs N. Some requirements:

1. We need to be able to estimate this curve for quantiles of predictive performance
2.  We need an estimate of uncertainty of our predictions along the curve. This will let the user know how certain the calculations are. We could also feed it into the algorithm so precision of the sample size estimate determines the number of simulations.
3. It needs to be computationally fast enough that the surrogate modelling doesn't stop the simulations.

**Comment by GForb** [comment]
## Ideas we currently have
1. Linear extrapolation (currently implemented). This is fine but we are not currently getting an estimate of uncertainty
2. Gausian process modelling - This seems to be the most promising approach. Tried and tested in Felix's work. 
3. Quantile regression (GAMS or otherwise) - This but we need to explore feasibility as we don't have experience.

Reply to this to add new ideas or comment on the ones we have:

**Comment by GForb** [comment]
## Implementation

We could offer different engines for surrogate modelling in the [same way as mlpw](R/fit.surrogate.R). This uses a switch statement to select which surrogate modelling to apply.

I am going to create a new branch and give this a go. Please put any thoughts on how this works below.

**Comment by flxzimmer** [comment]
### Estimate of uncertainty of our predictions

Did I get point (2) correctly that you refer to uncertainty in N? I struggled a lot with this question before and did not find a solution. What I fell back to was to concentrate on the uncertainty in the performance measure (both during the simulations and for reporting).

**Reply by GForb** [reply]
I don't think I did initially, but that is something I'd thought about.

As discussed today in the meeting I think we should concentrate on the uncertainty in the performance measure for now.
